languages/julia/exercises/concept/poetry-club/.meta/example.jl

```diff
@@ -0,0 +1,12 @@
+front_response(line) = line[begin]
```

Comment:
I'm not sure. I think `first` and `last` are slightly better since they'd also work on graphemes or other iterators.

One thing I want students to learn is that strings can be accessed via an index like arrays. Using `first`/`last` removes that aspect. That was also the idea behind the (now removed) third part. But then you run into the issue below:

```julia
julia> "Θεού"[2]
ERROR: StringIndexError("Θεού", 2)
```

Reply:
I don't think it's a good idea to index strings. We can teach them that strings are collections of characters if we contrive some reason to use map or collect or something?

We could do that as a run up to using `filter`? That's more efficient than repeatedly using `replace` anyway, I imagine.

Reply:
`filter` doesn't need `collect` apparently (I didn't know):

```julia
julia> filter(ispunct, s)
",,,,,,"
```

and `filter(!ispunct, s)` to do filter out punctuation as needed for this exercise works as well.

Reply:
That actually complicates things because I feel like higher-order functions are a concept themselves, and not necessarily one that seem like a natural prereq for a strings exercise.

Reply:
Yeah, I know about filter working direct on strings.

I meant that we could show what collect does, then what the predicates do, then what filter does or something.

But your right, higher order functions might be confusing.